### PR TITLE
DOC: Improvements in panel apply docstring

### PR DESCRIPTION
**Pull request by max-sixty**
Not sure what the bar is for a PR here - this is a mild but meaningful improvement - in particular re the option to supply numbers & a tuple - by no means perfect though

### CODE REVIEW

File: pandas/core/panel.py [hunk 1]
```diff
@@ -944,27 +944,28 @@ def construct_index_parts(idx, major=True):
...
         Examples
         --------
-        >>> p.apply(numpy.sqrt) # returns a Panel
-        >>> p.apply(lambda x: x.sum(), axis=0) # equiv to p.sum(0)
-        >>> p.apply(lambda x: x.sum(), axis=1) # equiv to p.sum(1)
-        >>> p.apply(lambda x: x.sum(), axis=2) # equiv to p.sum(2)
+        >>> panel.apply(np.sqrt) # returns a Panel
+        >>> panel.apply(lambda x: x.sum(), axis=1) # equivalent to p.sum(1), returning DataFrame
+        >>> panel.apply(lambda x: x.sum(), axis='minor') # equivalent to previous
+        >>> panel.apply(lambda x: x.shape, axis=(0,1)) # return the shapes of each DataFrame over axis 2, in Series
```
Comment (jreback) [inline pandas/core/panel.py]: to avoid going to long, add on a separate line (e.g. comment before), skipping a line between examples

**Comment by maximilianr** [comment]
@jreback green

File: pandas/core/panel.py [hunk 2]
```diff
@@ -944,27 +944,32 @@ def construct_index_parts(idx, major=True):
...
-        >>> p.apply(lambda x: x.sum(), axis=1) # equiv to p.sum(1)
-        >>> p.apply(lambda x: x.sum(), axis=2) # equiv to p.sum(2)
+	returns a Panel
+        >>> panel.apply(np.sqrt) 
+	equivalent to p.sum(1), returning a DataFrame
+        >>> panel.apply(lambda x: x.sum(), axis=1) 
+	equivalent to previous
+        >>> panel.apply(lambda x: x.sum(), axis='minor') 
+	return the shapes of each DataFrame over axis 2, as a Series
+        >>> panel.apply(lambda x: x.shape, axis=(0,1))
```
Comment (jorisvandenbossche) [inline pandas/core/panel.py]: I think there have to be blank lines between the code parts (`>>>`) and the explanations in between to render OK

(see '12. Examples' in https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#sections)

**Comment by maximilianr** [comment]
@jorisvandenbossche 
Thanks - I edited. I think we also can't just do `panel.apply`, but instead need the full expression `pd.Panel(pd.np.random.rand(4,3,2)).apply(pd.np.sqrt)`. Tell me if I'm wrong though

File: pandas/core/panel.py [hunk 3]
```diff
@@ -944,27 +944,40 @@ def construct_index_parts(idx, major=True):
...
         Additional keyword arguments will be passed as keywords to the function
 
         Examples
         --------
-        >>> p.apply(numpy.sqrt) # returns a Panel
-        >>> p.apply(lambda x: x.sum(), axis=0) # equiv to p.sum(0)
-        >>> p.apply(lambda x: x.sum(), axis=1) # equiv to p.sum(1)
-        >>> p.apply(lambda x: x.sum(), axis=2) # equiv to p.sum(2)
+
+	    returns a Panel with the square root of each element
```
Comment (jorisvandenbossche) [inline pandas/core/panel.py]: These lines should not have an indentation compared to the other lines

**Comment by jorisvandenbossche** [comment]
@maximilianr In theory, a full expression is not needed as this code is at the moment never run in order to build the docs (as is the case with the tutorials), but it is always nice to have full examples so people can copy paste it.

For layout, I think something like this is best (once define the variable, and then use it instead repeating the creation):

```
Returns a Panel with the square root of each element

>>> p = pd.Panel(np.random.rand(4,3,2))
>>> p.apply(np.sqrt)

Equivalent to p.sum(1), returning a DataFrame

>>> p.apply(lambda x: x.sum(), axis=1)

Equivalent to previous:

>>> p.apply(lambda x: x.sum(), axis='minor')

....
```

File: pandas/core/panel.py [hunk 4]
```diff
@@ -944,27 +944,40 @@ def construct_index_parts(idx, major=True):
...
+
+        >>> pd.Panel(pd.np.random.rand(4,3,2)).apply(pd.np.sqrt)
```
Comment (jorisvandenbossche) [inline pandas/core/panel.py]: Other comment: it is not needed to use `pd.np`, you can just use the common `np` numpy import

**Comment by maximilianr** [comment]
@jorisvandenbossche updated

File: pandas/core/panel.py [hunk 5]
```diff
@@ -944,27 +944,41 @@ def construct_index_parts(idx, major=True):
...
+
+        Equivalent to p.sum(1), returning a DataFrame
+
+        >>> p.apply(lambda x: x.sum(), axis=1)
+
+        Equivalent to previous:
+
+        >>> p.apply(lambda x: x.sum(), axis='minor')
+
+	    Return the shapes of each DataFrame over axis 2 (i.e the shapes of items x major), as a Series
```
Comment (jorisvandenbossche) [inline pandas/core/panel.py]: you just missed one here. For the rest looks very good!

**Comment by maximilianr** [comment]
@jorisvandenbossche I accidentally PEP8-ed the whole file. Shall I undo, or is that actually helpful?

**Comment by jreback** [comment]
@maximilianr yeh, normally want to do PEP stuff in separate PR's as confusing what is actually changed, but ok.

**Pull request merged**